media-api/app/lib/ImageExtras.scala

```diff
@@ -13,56 +13,59 @@ object ImageExtras {
+    "missing_description"         -> "Missing description *",
     "paid_image"                  -> "Paid imagery requires a lease",
     "over_quota"                  -> "The quota for this supplier has been exceeded",
     "conditional_paid"            -> "This image is restricted use"
   )
 
-  private def optToBool[T](o: Option[T]): Boolean =
-    o.map(some => some != false).getOrElse(false)
-
   def hasRights(rights: UsageRights) = rights match {
     case NoRights => false
     case _ => true
   }
-  def hasCredit(meta: ImageMetadata) = optToBool(meta.credit)
-  def hasDescription(meta: ImageMetadata) = optToBool(meta.description)
+  def hasCredit(meta: ImageMetadata) = !meta.credit.isEmpty
+  def hasDescription(meta: ImageMetadata) = !meta.description.isEmpty
 
   def hasCurrentAllowLease(leases: LeaseByMedia) =
-    optToBool(leases.current.map(_.access.name == "allow"))
+    leases.current.exists(_.access.name == "allow")
   def hasCurrentDenyLease(leases: LeaseByMedia) =
-    optToBool(leases.current.map(_.access.name == "deny"))
+    leases.current.exists(_.access.name == "deny")
 
-  import scala.concurrent.Await
-  import scala.util.Try
-  import scala.concurrent.duration._
-  import com.gu.mediaservice.lib.FeatureToggle
+  case class ValidityCheck(valid: Boolean, overrideable: Boolean, shouldOverride: Boolean) {
+    val isValid = valid && overrideable && shouldOverride
+  }
 
-  def validityMap(image: Image): Map[String, Boolean] = Map(
-    "paid_image"           -> Costing.isPay(image.usageRights),
-    "conditional_paid"     -> Costing.isConditional(image.usageRights),
-    "no_rights"            -> !hasRights(image.usageRights),
-    "missing_credit"       -> !hasCredit(image.metadata),
-    "missing_description"  -> !hasDescription(image.metadata),
-    "current_deny_lease"   -> hasCurrentDenyLease(image.leases),
-    "over_quota"           -> UsageQuota.isOverQuota(image.usageRights)
-  )
+  def validityMap(image: Image, withWritePermission: Boolean): Map[String, ValidityCheck] = {
+    val shouldOverride = validityOverrides(image, withWritePermission).exists(_._2 == true)
+
+    def createCheck(validCheck: Boolean, overrideable: Boolean = true) =
+      ValidityCheck(validCheck, overrideable, shouldOverride)
+
+    Map(
+      "paid_image"           -> createCheck(Costing.isPay(image.usageRights)),
+      "conditional_paid"     -> createCheck(Costing.isConditional(image.usageRights)),
+      "no_rights"            -> createCheck(!hasRights(image.usageRights)),
+      "missing_credit"       -> createCheck(!hasCredit(image.metadata), false),
+      "missing_description"  -> createCheck(!hasDescription(image.metadata), false),
+      "current_deny_lease"   -> createCheck(hasCurrentDenyLease(image.leases)),
+      "over_quota"           -> createCheck(UsageQuota.isOverQuota(image.usageRights))
+    )
+  }
 
   def validityOverrides(image: Image, withWritePermission: Boolean): Map[String, Boolean] = Map(
```

Comment:
I think `validityOverrides` should move up to be above where `shouldOverride` is defined then it makes it a bit easier to follow the logic of createCheck method. Or it should be inlined with `shouldOverride` (as that is a val so called just once).